spec/service_models/miq_ae_service_service_spec.rb

```diff
@@ -72,6 +72,20 @@ def invoke_ae
 
         @parent.reload
         expect(@parent.direct_service_children.collect(&:id)).to eq([@service.id])
+        expect(@parent.service_resources.collect(&:resource_id)).to eq([@service.id])
+        expect(@parent.service_resources.first).to be_a_kind_of(ServiceResource)
+      end
+
+      it "raises error if child service is already owned by different parent" do
+        @parent1 = FactoryGirl.create(:service, :name => "parent_service1")
+        method = "$evm.root['service'].parent_service = $evm.vmdb('service').find(#{@parent1.id})"
+        @ae_method.update_attributes(:data => method)
+        invoke_ae
+
+        method = "$evm.root['service'].parent_service = $evm.vmdb('service').find(#{@parent.id})"
+        @ae_method.update_attributes(:data => method)
+
+        expect { invoke_ae }.to raise_error(MiqAeException::AbortInstantiation)
```

Comment:
This test could be written to use the service model directly without having to call `invoke_ae` which would be quicker and simpler while still validating the same use case: 

Something like:
```ruby
      it "raises error if child service is already owned by different parent" do
        parent1 = FactoryGirl.create(:service)
        service.add_to_service(parent1)
        parent_service = MiqAeMethodService::MiqAeServiceService.find(@parent.id)

        expect { service_service.parent_service = parent_service }.to raise_error(MiqException::Error)
      end
```